src/mpid/common/genq/mpidu_genq_shmem_queue.h

```diff
@@ -99,16 +99,9 @@ static inline int MPIDU_genq_nem_enqueue(MPIDU_genq_shmem_queue_t queue, void *c
 
 /* INVERSE QUEUE */
 
-static inline void *MPIDU_genq_shmem_queue_head(MPIDU_genq_shmem_queue_t queue);
```

Comment:
We need to keep these interfaces. SHM collectives is going to need these.

Reply:
With MPSC queue, the queue head access needs thread safety. So having this API, in addition to needed implementation for each queue (does it even require implementation from each such as serial queue), it will require enqueue code be placed inside the same critical section. 

Can we add this API at the time that we are going to use it? Then we can properly review and balance the complexity and performance concerns.

A separate question: why do we need serial queue? What is the planned use case?

Reply:
Pending @yfguo